index.js

```diff
@@ -89,66 +91,107 @@ async function runEveryMinute({ global, storage, cache }) {
-                            invoice_date: new Date(upcomingInvoiceDate).toLocaleDateString('en-GB'),
-                            invoice_current_amount: upcomingInvoice.amount_due / 100,
-                            distinct_id: customer.email || customer.id
-                        })
-                        await cache.set(`last_invoice_${customer.id}`, upcomingInvoiceDate)
-                    }
+                const upcomingInvoiceResponse = await fetchWithRetry(
+                    `https://api.stripe.com/v1/invoices/upcoming?customer=${customer.id}`,
+                    global.defaultHeaders
+                )
+                const upcomingInvoice = await upcomingInvoiceResponse.json()
+
+                const upcomingInvoiceDate = upcomingInvoice.created * 1000
+
+                const invoiceData = {
+                    product: productName,
+                    amount_due: upcomingInvoice.amount_due / 100,
+                    customer: customer.email,
+                    quantity: 0
+                }
+                upcomingInvoice.lines?.data.forEach((line) => {
+                    invoiceData.quantity += line.quantity
+                })
+
+                if (productName in invoicesByProduct) {
+                    invoicesByProduct[productName].push(invoiceData)
+                } else {
+                    invoicesByProduct[productName] = [invoiceData]
+                }
+
+                if (!upcomingInvoice.error && upcomingInvoice.created) {
+                    posthog.capture('Upcoming Invoice', {
+                        amount: invoiceData.amount_due,
+                        invoice_date: new Date(upcomingInvoiceDate).toLocaleDateString('en-GB'),
+                        product: invoiceData.product,
+                        quantity: invoiceData.quantity,
+                        stripe_customer_id: customer.id,
+                        distinct_id: customer.email || customer.id,
+                        $set: customer.email ? { email: customer.email } : undefined
+                    })
                 }
             }
         }
 
-        const customerRecordExists = await storage.get(customer.id)
+        // Note: I think if we want to automatically update users we should just update properties instead of firing events
+        // I also think we should perhaps restrict the properties we send so it doesn't feel overwhelming to
+        // see a lot of properties you may not understand.
+
+        // const customerRecordExists = await storage.get(customer.id)
+        //
+        // if (!customerRecordExists) {
+        //     await storage.set(customer.id, true)
+        // }
+        //
+        // if (global.onlyRegisterNewCustomers && customerRecordExists) {
+        //     break
+        // }
+        //
+        // posthog.capture(customerRecordExists ? 'update_stripe_customer' : 'new_stripe_customer', {
+        //     ...properties,
+        //     $set: basicProperties
+        // })
+    }
```

Comment:
Used this section for some questions, but otherwise let's remove the comments - no need to have this lying around